src/NHibernate/Async/Criterion/QueryOver.cs

```diff
@@ -0,0 +1,69 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+using NHibernate.Criterion.Lambda;
+using NHibernate.Engine;
+using NHibernate.Impl;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+
+namespace NHibernate.Criterion
+{
+	using System.Threading.Tasks;
+	public abstract partial class QueryOver<TRoot> : QueryOver, IQueryOver<TRoot>
+	{
+
+		private async Task<IList<TRoot>> ListAsync()
+		{
+			return await (criteria.ListAsync<TRoot>()).ConfigureAwait(false);
```

Comment:
[Fixed](https://github.com/maca88/AsyncGenerator/commit/d9b8b5b1fe021aa71ea7e25972a31d06da5803dc). The problem lied in the comparison of the return types when there were type parameters invloved. For type parameters a simple `Equals` is not enough to check if they matches.